### PR TITLE
#214 fix offsets assigments

### PR DESCRIPTION
**Pull request by artpaul**

### CODE REVIEW

File: dbms/include/DB/Functions/FunctionsStringSearch.h [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ struct ReplaceStringImpl
 			bool can_finish_current_string = false;
 
 			/// Проверяем, что вхождение не переходит через границы строк.
-			if (match + needle.size() < begin + n * (i + 1))
+			if (match + needle.size() - 1 < begin + n * (i + 1))
```
Comment (artpaul) [inline dbms/include/DB/Functions/FunctionsStringSearch.h]: This is because match points to first symbol from needle.

Comment (alexey-milovidov) [inline dbms/include/DB/Functions/FunctionsStringSearch.h]: Could you also provide a test?

Comment (artpaul) [inline dbms/include/DB/Functions/FunctionsStringSearch.h]: Yes, but could you please remember me how to run tests locally?

Comment (alexey-milovidov) [inline dbms/include/DB/Functions/FunctionsStringSearch.h]: Put `clickhouse` program to `/usr/bin` (or PATH);
make symlink `clickhouse-client` in `/usr/bin` pointing to `clickhouse`*.

Run clickhouse-server.

Change directory to `dbms/tests/` in source tree
and run `./clickhouse-test`.

*- Also you could just build (`./release --standalone`) and install debian packages.

Comment (artpaul) [inline dbms/include/DB/Functions/FunctionsStringSearch.h]: Done.

**Comment by alexey-milovidov** [comment]
Thanks!

**Pull request merged**